src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -122,20 +125,35 @@ public Thread newThread(Runnable r) {
     };
   }
 
-  private Runnable farmerTask() {
-    return new Runnable() {
-      @Override
-      public void run() {
-        log.debug("Starting farmer task");
-        try {
-          while (!stopRequested) {
-            submitBatchWhenReady();
+  // visible for testing
+  Runnable farmerTask() {
+    return () -> {
+      log.debug("Starting farmer task");
+      try {
+        List<Future<BulkResponse>> futures = new ArrayList<>();
+        while (!stopRequested) {
+          // submitBatchWhenReady waits for lingerMs so we won't spin here unnecessarily
+          futures.add(submitBatchWhenReady());
+
+          // after we submit, look at any previous futures that were completed and call get() on
+          // them so that exceptions are propagated.
+          List<Future<BulkResponse>> unfinishedFutures = new ArrayList<>();
+          for (Future<BulkResponse> f : futures) {
+            if (f.isDone()) {
+              log.debug("Bulk request completed with status {}", f.get());
+              f.get();
```

Comment:
you can remove this second `get()`.

Reply:
I'm unsure if the debug statements would get hit if log level is not at least DEBUG, so I left both in. Yeah, that's a bit confusing. Let me write that a little better.

Reply:
`f.get()` will get evaluated always. the only thing that we will avoid with the debug level is the string formatting/concatenation.